lab/skel_graph/traversal/problems/alee/alee.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 Nicolae Mara-Ioana <maraioana9967@gmail.com>
+
+// Alee - Drumuri minime in graf
+// Complexitate: O(n * m)
+
+// https://www.infoarena.ro/problema/alee
+
+#include <bits/stdc++.h>
+
+class Task {
+	
+ public:
+	
+	void solve() {
+		read_input();
+		print_output(get_result());
+	
+	}
+
+ private:
+	int n;
+	int m;
+	int source;
+	std::vector<std::vector<char>> graph;
+    std::pair<int, int> startGate;
+    std::pair<int, int> endGate;
+	
+    // s-a ales urmatoarea codificare:
+    // graph[i][j] = 'T', atunci exista un copcac la coordonatele i, j
+    // graph[i][j] = ' ', atunci exista zona libera la coordonatele i, j
+	void read_input() {	
+        std::cin >> n >> m;
+        // initial se marcheaza toate elementele din matrice ca fiind zone libere
+        graph = std::vector<std::vector<char>>(n + 1, 
+            std::vector<char>(n + 1, ' '));
+		for (int i = 1, x, y; i <= m; i++) {
+            // se citesc coordonatele la care se afla copacii
+            // se actualizeaza valorile din celulele in care se regasesc obstacole
+			std::cin >> x >> y;
+			graph[x][y] = 'T';
+		}
+
+        // Se citesc coordonatele portii din care se porneste cautarea, 
+        // si cele ale portii ce reprezinta destinatia
+        std::cin >> startGate.first >> startGate.second 
+            >> endGate.first >> endGate.second;
+	}
+	
+    // intrucat graful este reprezentat sub forma unei matrice
+    // se va aplica Algoritmul lui Lee
+	int get_result() {
+        // se tine evidenta nodurilor vizitate, 
+        // initial toate nodurile sunt nevizitate
+		std::vector<std::vector<bool>> visited(n + 1, 
+            std::vector<bool>(n + 1, false));
+
+        // vectorul de distante pornind de la poarta de start
+		std::vector<std::vector<int>> minPath(n + 1, 
+            std::vector<int>(n + 1, INT_MAX));
+
+        // in coada se vor retine nodurile descoperite, dar inca nefinalizate
+		std::queue<std::pair<int, int>> nodeQueue;
+
+        // vectorul ce reprezinta posibilele pozitii 
+        // ale vecinilor unui element din matrice
+        std::vector<std::pair<int, int>> directions;
```

Comment:
We can declare this as `const` and initialization with `{ {1, 0}, {0, 1}, {-1, 0}, {0, -1}}`